resource.go

```diff
@@ -141,11 +142,11 @@ func ensureDefaultPatches(clusterStatus providerv1alpha1.StatusCluster, patches
 	nodesEmpty := clusterStatus.Nodes == nil
 	versionsEmpty := clusterStatus.Versions == nil
 
-	if conditionsEmpty && nodesEmpty && versionsEmpty {
+	if reflect.DeepEqual(clusterStatus, providerv1alpha1.StatusCluster{}) {
 		patches = append(patches, Patch{
 			Op:    "add",
-			Path:  "/status",
-			Value: Status{},
+			Path:  "/status/cluster",
```

Comment:
Is this backward compatible?

Reply:
It's not, and I think intention is to break backward compatibility with fix for what was a bug - clearing `/status` as part of initializing `/status/cluster`. Whatever if anything is relying on this invalid logic will have to be fixed.

Thank you very much @rossf7 for verifying this change works for https://github.com/giantswarm/kvm-operator/pull/804 and https://github.com/giantswarm/azure-operator/pull/617, in addition to e2e tests in https://github.com/giantswarm/aws-operator/pull/2090 where the bug was uncovered.